executor/brie.go

```diff
@@ -43,24 +52,64 @@ import (
 	"github.com/pingcap/tidb/sessionctx/variable"
 	"github.com/pingcap/tidb/types"
 	"github.com/pingcap/tidb/util/chunk"
+	"github.com/pingcap/tidb/util/logutil"
 	"github.com/pingcap/tidb/util/sqlexec"
+	pd "github.com/tikv/pd/client"
+	"go.uber.org/zap"
+)
+
+const (
+	defaultImportID = "tidb_import_"
+
+	// current commands/status that displayed in progress
+	// there're still some commands didn't list here, whereas they are used as arguments of StartProgress in BR
+	cmdWait        = "Wait"
+	cmdBackup      = "Backup"
+	cmdRestore     = "Restore"
+	cmdImport      = "Import"
+	cmdPostProcess = "Post-process"
+	cmdStop        = "Stop"
+	cmdFinish      = "Finish"
+)
+
+var (
+	// CreateSessionForBRIEFunc is used to solve cycle import when use session.CreateSession
+	CreateSessionForBRIEFunc func(sessionctx.Context) (checkpoints.Session, error)
 )
 
 // brieTaskProgress tracks a task's current progress.
 type brieTaskProgress struct {
-	// current progress of the task.
-	// this field is atomically updated outside of the lock below.
+	// cmd is the name of the step the BRIE task is currently performing.
+	// We didn't keep it consistent with below two numeric fields. That is to say, user may see
+	// Import 99% -> Import 100% -> Post Process 100% , where 100% always means "Post Process"
+	// To avoid concurrency problems, always use getter/setter
+	cmd atomic.Value
+
+	// current progress of the task. When update current merely, we could use atomic operation.
+	// When update current according to total, we should lock total first
 	current int64
 
-	// lock is the mutex protected the two fields below.
+	// lock protects total. In some scenarios, current is decided by total, so locking total first could also keep
```

Comment:
do we need to define this as
lock struct{
    current int64
    total int64
}